pennylane/math/multi_dispatch.py

```diff
@@ -376,11 +378,12 @@ def tensordot(tensor1, tensor2, axes=None):
     Returns:
         tensor_like: the tensor product of the two input tensors
     """
-    interface = _multi_dispatch([tensor1, tensor2])
-    return np.tensordot(tensor1, tensor2, axes=axes, like=interface)
+    tensor1, tensor2 = np.coerce([tensor1, tensor2], like=like)

```

Comment:
This is minor, but do you think we should include `coerce` inside the decorator? Or is the required behaviour to specific to different functions?

The reason I suggest this is we could simplify even more, and just do

```python
tensordot = multi_dispatch(argnum=[0, 1])(np.tensordot)
```

Reply:
I was thinking about this as well :laughing: Maybe as an optional kwarg?
```python
tensordot = multi_dispatch(argnum=[0, 1], do_coerce=True)(np.tensordot)
```

Reply:
I like it 😎 Although, I might even just have `coerce=True`?

Reply:
Discussed off-PR: Including this in `multi_dispatch` is less straight-forward than previously thought, due to unpacking and re-packing of arguments in that are sequences of tensors. Therefore it will not be included.